libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.types.ts

```diff
@@ -0,0 +1,50 @@
+import type {
+  CertificateRewardDistributionOutput,
+  RewardsDistributionActorType,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+import type {
+  MethodologyActorType,
+  NonEmptyString,
+} from '@carrot-fndn/shared/types';
+import type BigNumber from 'bignumber.js';
+
+export interface RuleSubject {
+  creditUnitPrice: BigNumber;
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[];
+}
+
+export interface ResultContentsWithMassIdCertificateValue {
+  massIdCertificateValue: BigNumber;
+  resultContent: CertificateRewardDistributionOutput;
+}
+
+export interface Participant {
+  id: NonEmptyString;
+  name: NonEmptyString;
+}
+
+export interface RewardsDistributionActor {
+  actorType: MethodologyActorType.REMAINDER | RewardsDistributionActorType;
+  amount: NonEmptyString;
+  participant: Participant;
+  percentage: NonEmptyString;
+}
+
+export type ActorsByActorType = Map<string, RewardsDistributionActor>;
+
+export interface RewardsDistribution {
+  actors: RewardsDistributionActor[];
+  certificateTotalValue: NonEmptyString;
+  creditsUnitPrice: NonEmptyString;
```

Comment:
Just nitpicking here 😬 
```suggestion
  creditUnitPrice: NonEmptyString;
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/222/commits/609e88089d58949b4e508881d0cbaa55d6106435